### PR TITLE
Enable auth validate passwords

### LINKED ISSUE #5551: Local instance allows weak passwords
## Details

The [installation instructions](https://docs.readthedocs.io/en/stable/install.html) say to run `python manage.py createsuperuser`, which prompts for a password; self-sign-up users are also prompted for a password. By default, `createsuperuser` requires a non-empty password, and self-sign-up requires a password that is six characters or longer. Neither impose any further password complexity requirements, including checking for well-known passwords like `password` or `123456`.

## Expected Result

An error message - or at least a warning - about using insecure passwords.

## Actual Result

"Superuser created successfully." and successful sign-up.

### PR DESCRIPTION
**Pull request by humitos**
Define some simple validators to use when creating a new user or
changing the password.

https://docs.djangoproject.com/en/1.11/topics/auth/passwords/#module-django.contrib.auth.password_validation

Closes #5551

### CODE REVIEW

File: readthedocs/settings/base.py [hunk 1]
```diff
@@ -181,6 +181,24 @@ def USE_PROMOS(self):  # noqa
         'allauth.account.auth_backends.AuthenticationBackend',
     )
 
+    AUTH_PASSWORD_VALIDATORS = [
+        {
+            'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
+        },
+        {
+            'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
```
Comment (ericholscher) [inline readthedocs/settings/base.py]: I really dislike these defaults in dev. I think these should likely only be prod settings.

**Comment by davidfischer** [comment]
I'm in favor of these settings being present in the base settings file as they are in this PR. This helps others who take our code to do the right thing as well as ensuring that all the different instances we control (community site, corporate site, etc.) have sound security defaults.

I can understand @ericholscher's position and we could also set `AUTH_PASSWORD_VALIDATOR=[]` in `dev.py` to ease the development settings.

File: readthedocs/settings/base.py [hunk 2]
```diff
@@ -181,6 +181,24 @@ def USE_PROMOS(self):  # noqa
...
+            'OPTIONS': {
+                'min_length': 9,
```
Comment (davidfischer) [inline readthedocs/settings/base.py]: Was 9 characters a recommendation from somewhere? I believe 8 is more common.

Comment (davidfischer) [reply readthedocs/settings/base.py]: I see. It was copied from the Django docs.

**Comment by humitos** [comment]
> I can understand @ericholscher's position and we could also set `AUTH_PASSWORD_VALIDATOR=[]` in `dev.py` to ease the development settings.

I like this. I will update the PR for this.

**Pull request merged**